### ISSUE TITLE
Geochart: colorAxis doesn't show gradient when <base> tag is set

### ISSUE THREAD

**Issue by g3852477** [google/google-visualization-issues#2000]
Hi there,

this is a new ticket for Issue 598, since it was marked as closed, but it was not completely fixed and there have been no reaction to my post since a week.
https://github.com/google/google-visualization-issues/issues/598

See here for details:
https://github.com/google/google-visualization-issues/issues/598#issuecomment-111475593

The issue is mentioned at stackoverflow:
http://stackoverflow.com/questions/29328092/google-charts-geochart-coloraxis-doesnt-show-gradient

Short description:
When <base> tag in head section is set, no gradient is shown at GeoChart. The suggested workaround doesn't fix the issue completely.

Here is the of copy my text from issue 598:

The issue is reproducible in Chrome 42.0.2311.152 and Firefox 38.0.5 (latest) in Windows 7

Below is a short example (full HTML file) to reproduce.
GitHub escaped special characters in HTML below so please replace 
&amp;quot; with "
&amp;#39; with '
&amp;lt; with <
&amp;gt; with >

Any comments and helps regarding the issue is appreciated! 
PS Would be nice, if google fixes such bugs in their code, so we don't need any strange work-around. 

&lt;!DOCTYPE html&gt;&lt;html&gt;&lt;head&gt;
&lt;base href=&quot;http://www.google.com/bugs&quot; /&gt;
&lt;script src=&quot;http://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js&quot;&gt;&lt;/script&gt;
&lt;script type=&quot;text/javascript&quot; src=&#39;https://www.google.com/jsapi?autoload={&quot;modules&quot;:[{&quot;name&quot;:&quot;visualization&quot;,&quot;version&quot;:&quot;1&quot;,&quot;packages&quot;:[&quot;geochart&quot;]}]}&#39;&gt;&lt;/script&gt;
&lt;script type=&quot;text/javascript&quot;&gt;
google.setOnLoadCallback(function()
{
    data = google.visualization.arrayToDataTable([ 
            [ &quot;Code&quot;, &quot;Location&quot;, &quot;Votes&quot; ], [ &quot;NL-DR&quot;, &quot;Drenthe&quot;, 790 ], [ &quot;NL-FL&quot;, &quot;Flevoland&quot;, 196 ],
            [ &quot;NL-FR&quot;, &quot;Fryslan&quot;, 632 ], [ &quot;NL-GE&quot;, &quot;Gelderland&quot;, 776 ], [ &quot;NL-GR&quot;, &quot;Groningen&quot;, 609 ], [ &quot;NL-LI&quot;, &quot;Limburg&quot;, 919 ],
            [ &quot;NL-NB&quot;, &quot;Noord-Brabant&quot;, 726 ], [ &quot;NL-NH&quot;, &quot;Noord-Holland&quot;, 56 ], [ &quot;NL-OV&quot;, &quot;Overijssel&quot;, 890 ], [ &quot;NL-UT&quot;, &quot;Utrecht&quot;, 712 ],
            [ &quot;NL-ZE&quot;, &quot;Zeeland&quot;, 888 ], [ &quot;NL-ZH&quot;, &quot;Zuid-Holland&quot;, 465 ], ]);

```
var options = {
    width : 500,
    height : 500,
    region : &quot;NL&quot;,
    dataMode : &quot;regions&quot;,
    resolution : &quot;provinces&quot;,
};

function fixGoogleCharts(divId)
{
    return function()
    {
        $(&quot;svg&quot;, &quot;#&quot; + divId).each(function()
        {
            $(this).find(&quot;g&quot;).each(function()
            {
                if ($(this).attr(&quot;clip-path&quot;))
                {
                    if ($(this).attr(&quot;clip-path&quot;).indexOf(&quot;url(#&quot;) == -1)
                        return;
                    $(this).attr(&quot;clip-path&quot;, &quot;url(&quot; + document.location + $(this).attr(&quot;clip-path&quot;).substring(4))
                }
            });
            $(this).find(&quot;rect&quot;).each(function()
            {
                if ($(this).attr(&quot;fill&quot;))
                {
                    if ($(this).attr(&quot;fill&quot;).indexOf(&quot;url(#&quot;) == -1)
                        return;
                    $(this).attr(&quot;fill&quot;, &quot;url(&quot; + document.location + $(this).attr(&quot;fill&quot;).substring(4))
                }
            });
        });
    }
}

var divName = &quot;regions_div&quot;;
var chart = new google.visualization.GeoChart(document.getElementById(divName));
google.visualization.events.addListener(chart, &quot;ready&quot;, fixGoogleCharts(divName));
chart.draw(data, options);
```

});
&lt;/script&gt;
&lt;/head&gt;
&lt;body&gt;&lt;div id=&quot;regions_div&quot;&gt;&lt;/div&gt;&lt;/body&gt;&lt;/html&gt;

**Comment by dlaliberte** [comment]
This should be fixed in the next release of Google Charts.  Since browsers are changing their behavior regarding the SVG url references in the presence of the BASE tag, we have to adapt.  In the next release, only IE9 is excluded from making the urls absolute.

**Comment by g3852477** [comment]
Hi dlaliberte,

thanks for quick answer!

Could you please also comment, when a new release of Google Charts is expected, or can be expected? 
This is an important information for us, because this bug prevents us from using the current version of GeoChart.

Regards,
g3852477

**Comment by dlaliberte** [comment]
The next release is still blocked on making the previous release available in frozen form.  We are getting closer, but I still can't say exactly when we will be done.

**Comment by orwant** [comment]
If it puts your mind at ease, we aim for a release every couple of months.

Jon

**Comment by g3852477** [comment]
thanks for details, dlaliberte and Jon (orwant)!
Looking forward to getting the new release.

**Comment by g3852477** [comment]
@orwant @dlaliberte 
Hi there guys, 
any news about the next release of Google Charts? 
I'm looking forward to getting the fix for a geochart, when colorAxis doesn't show gradient when <base> tag is set.

**Comment by grabks** [comment]
@g3852477 We have been having some issues with pushing the current release. In the meanwhile, you can use [frozen versions](https://developers.google.com/chart/interactive/docs/library_loading_enhancements#frozen-versions) to access version 43.

**Comment by g3852477** [comment]
@grabks Thanks for the info! I tested version 43 and the bug is not reproducible. 
Do you have a changelog of frozen versions, where the changes and their dates are specified?
It would be also nice to know the estimation of the date of the official release.
Thanks again.

**Comment by orwant** [comment]
https://developers.google.com/chart/interactive/docs/release_notes

Jon
